contributors/design-proposals/service_links.md

```diff
@@ -0,0 +1,237 @@
+# Explicit service links
+
+## Abstract
+
+Proposes an API for explicit links to services in the pod spec.
+
+## Motivation
+
+Currently, all pods are automatically injected with docker-link-style
+environment variables for each service in their namespaces.  Making service
+dependencies explicit in the PodSpec has a number of advantages:
+
+1.  Reduce accidental coupling
+2.  Reduce the number of IP tables rules that have to be created by the
```

Comment:
I think the effort to lazy-create iptables rules is more energy than it is worth.  I don't mind thinking about it, but it's somewhat tricky and maybe not very fruitful.  We have to handle more than just "does any pod need it" but also external traffic, which we can't know a priori.  With the local-only extension, we can reduce scope, but any Service that isn't using that needs to be present on every node.

This should not be a motivating design point, IMO.